packages/core/src/Element.ts

```diff
@@ -61,69 +70,82 @@ export const Element = {
 
     return {
       type: "Element",
-      value,
-      hierarchy,
+      value: Promise.resolve(value),
+      hierarchy: Promise.resolve(hierarchy),
```

Comment:
from discussion, @tjjfvi suggested using `Promisish` for type as done elsewhere and removing `Promise.resolve`